NYPL-discovery/discovery-front-end
# MEGA PR: Research Catalog changes - hero, subnav, breadcrumbs, and search bar changed everywhere

**What's this do?**
This combines all the work from PRs #1654, #1651, #1647, #1645. That way all the changes can be run locally at once.

This PR introduces the new banner design (Breadcrumb, Heading, Subnav) and new Search Bar on landing and search page. There are a lot of wrapping `<div>`s and CSS removed.

[Figma](https://www.figma.com/file/CCCdFt8H7pMqHjkRQlivSw/IA---Templates-%2B-Components?node-id=44960%3A79131)

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/SCC-2478 - "Apply hero to all non-shep SCC pages"
https://jira.nypl.org/browse/SCC-2539 - "Search Result Page - search bar and refine result"
https://jira.nypl.org/browse/SCC-2451 - "Breadcrumbs"
https://jira.nypl.org/browse/SCC-2453 - "Subnav"
https://jira.nypl.org/browse/SCC-2555 - "Update SHEP pages with new designs"

**Do these changes have automated tests?**
Had to adjust tests as needed. Did not add new tests.

**How should this be QAed?**
Make sure `.env` file has any new variables - `ITEM_BATCH_SIZE`, `DISPLAY_TITLE`
If we are still dealing with the Patron Service QA bug, try using prod Platform API.

Hold Confirmation page can be seen here: http://local.nypl.org:3001/research/collections/shared-collection-catalog/hold/confirmation/b12741387-i31190923
- There is a bug on the above page that the item title does not display when the page is hit directly. This page is inherently weird though, because it can be displayed without an actual hold being placed.

**Dependencies for merging? Releasing to production?**
For production - The Notification component needs updating and so does Subject Heading Search.

**Has the application documentation been updated for these changes?**

**Did someone actually run this code to verify it works?**
I did
